src/patientControl/PatientSelectionModal.jsx

```diff
@@ -0,0 +1,164 @@
+                onMouseLeave={() => { this.setState({ hovered: null }); }}
+                onClick={() => { this.switchPatients(listItem.shrId); }}
+            >
+                <ListItemText primary={listItem.time} className='app-time' />
+                <ListItemIcon><img alt='' src={listItem.pic} style={{ width: '100px', height: '100px', marginRight: '0px' }}></img></ListItemIcon>
+                <ListItemText primary={listItem.name} secondary={listItem.description} className='modal-description' />
+            </ListItem>);
+        });
+    }
+
+    switchPatients = (shrId) => {
+        this.handleClose();
+        this.props.loadPatient(shrId);
+    }
+
+    buildPatientDescription = (encounter, patient) => {
+        let timeSinceLast = 'first visit';
+        const encounterMom = new moment(encounter.expectedPerformanceTime, "DD MMM YYYY HH:mm");
+        if (patient.getPreviousEncounter() !== undefined) {
+            const lastSeen = new moment(patient.getPreviousEncounter().expectedPerformanceTime, "DD MMM YYYY HH:mm");
+            timeSinceLast = 'last seen: ' + lastSeen.fromNow();
+        }
+        if (encounterMom.isAfter(moment().subtract(1, 'h')) && encounterMom.isBefore(moment())) { //if the appointment started less than an hour ago
+            timeSinceLast = 'in progress: began ' + encounterMom.fromNow();
+        }
+        // For future implementation of "note started" feature:
+        // const noteStarted = '3 hours';
+        // let description = 'last seen: ' + timeSinceLast + ' \n note started ' + noteStarted + ' ago';
+        // const secondary = description.split('\n').map((item, i) => <p key={i} style={{ margin: '0px' }}>{item}</p>);
+        return timeSinceLast;
+    }
+
+    handleClose = () => {
+        this.props.handleClose();
+    }
+
+    getModalStyle() {
+        const top = 50;
+        const left = 50;
+        return {
+            top: `${top}%`,
+            left: `${left}%`,
+            transform: `translate(-${top}%, -${left}%)`,
+            position: 'absolute',
+            width: 400,
+            backgroundColor: 'white',
+            boxShadow: 'black',
+            padding: 8,
+        };
+    }
+
+    render() {
+        const { isTablet } = this.props;
+        if (isTablet) {
+            return (<Modal
+                aria-labelledby='simple-modal-title'
+                open={this.props.isModalOpen}
+                onClose={this.handleClose}
+            >
+                <div style={this.getModalStyle()} className='modal'>
+                    <FontAwesome className='fas fa-times clickable' name='close-icon' onClick={this.handleClose} />
+                    <List>
+                        {this.fillModal()}
+                    </List>
+                </div>
+            </Modal>);
+        } else {
+            return <div></div>;
+        }
+    }
+}
+PatientSelectionModal.propTypes = {
+    isModalOpen: PropTypes.bool.isRequired,
```

Comment:
[JA4]: I think there are additional props you added to this component that didn't make it to the propTypes here.

Reply:
These look good too! Are the last two also required like the others?

Reply:
yup! resolved!